Topic1/src/ShoppingCart.java

```diff
@@ -0,0 +1,119 @@
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class ShoppingCart implements Subject{
+	private String IdCart;
+	private List<IOffer> items = new ArrayList<IOffer>();
+	
+	private IPayStrategy strategy; //Pay Strategy
+	
+	private List<Observer> observers = new ArrayList<Observer>();
+
+	private Sequence counter = Sequence.getInstance();
+	
+	public ShoppingCart(String id,MarketManager m){
+		setIdCart(id);
+		addObserver(m.getInstance()); //add observer 
```

Comment:
Warning: Since it is an static call we don't need to pass the MarketManager as a parameter. We could simply do a MarketManager.getInstance();